src/main.c

```diff
@@ -82,6 +85,49 @@ bool parse_rect(const char* arg, struct rect* rect)
     return true;
 }
 
+char** read_filelist(size_t* files_num)
+{
+    size_t buffer_len = PATH_MAX;
+    char* buffer = NULL;
+    ssize_t line_len;
+
+    *files_num = 0;
+    size_t files_max = 32;
+    char** files = malloc(sizeof(char*) * files_max);
+    if(!files) {
+        fprintf(stderr, "Not enough memory!\n");
+        return NULL;
+    }
+
+    // set stdin to non-blocking, so it doesn't hang waiting for interactive input
+    int flags = fcntl(0, F_GETFL, 0);
+    fcntl(0, F_SETFL, flags | O_NONBLOCK);
+
+    line_len = getline(&buffer, &buffer_len, stdin);
```

Comment:
1. It's a corner-case, rather rare one. The usual way to deal with this problem is to introduce an option to specify null separated data, e.g.: `find -print0 | xargs -0 ...` It is never default behaviour to use null terminated list, because all common unix tools produce newline separated data and it is much more convenient.
2. The application certainly has right to define its "API" and here we say the data must be separated by `\n`. Or by null byte, if that option would be implemented and used.